cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaTypes.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * Contains types used in parsing an avro schema.
+ * TODO: This is a work in progress. We don't use these types yet fully
+ * in the schema parser yet.
+ */
+type IComplexTypeNames = 'array' | 'enum' | 'map' | 'record' | 'union';
+type ISimpleType =
+  | 'boolean'
+  | 'bytes'
+  | 'date'
+  | 'decimal'
+  | 'double'
+  | 'float'
+  | 'int'
+  | 'long'
+  | 'number'
+  | 'string'
+  | 'time'
+  | 'timestamp';
+type ILogicalTypeNames = 'timestamp-micros' | 'date' | 'time-micros' | 'decimal';
+
+type IDisplayType = ISimpleType | IComplexTypeNames;
+
+type ISimpleTypeNullable = Array<ISimpleType | 'null'>;
```

Comment:
Should this be null (without the single quotes)?

Reply:
No. `'null'` should be a string. `'null'` in avro is actually a valid type. We treat it as auxillary information for any type because it makes user experience much simpler.